### PR TITLE
Parameterize version and item tests

### PR DESCRIPTION
**Pull request by bigtedde**
Continued parameterization efforts in the the test suite. This draft PR focuses on the `version` and `item` tests.

### CODE REVIEW

File: src/borg/testsuite/version.py [hunk 1]
```diff
@@ -30,15 +30,10 @@ def test_parse_version(version_str, version_tuple):
     assert parse_version(version_str) == version_tuple
 
 
-def test_parse_version_invalid():
+@pytest.mark.parametrize("invalid_version", ["", "1", "1,2", "crap"])
```
Comment (ThomasWaldmann) [inline src/borg/testsuite/version.py:33]: the original test has `1.2`, not `1,2`.

File: src/borg/testsuite/item.py [hunk 2]
```diff
@@ -158,19 +157,19 @@ def test_item_optr():
...
+    [
+        ([b"1234", b"567A", b"bC"], [b"1", b"23", b"4567A", b"b", b"C"], True),  # equal
+        ([b"12345"], [b"1234", b"56"], False),  # one iterator exhausted before the other
+        ([b"1234", b"65"], [b"1234", b"56"], False),  # content mismatch
+        ([b"1234", b"56"], [b"1234", b"565"], False),  # first is the prefix of second
+    ],
+)
+def test_chunk_content_equal(chunk_a, chunk_b, chunks_equal):
+    chunks_a = [data for data in chunk_a]
+    chunks_b = [data for data in chunk_b]
```
Comment (ThomasWaldmann) [inline src/borg/testsuite/item.py:171]: while we clean this up, maybe we could do a small readability improvement:

- have the test parameter lists with str members (no `b` for bytes)
- use `.encode()` to create the lists with bytes objects in 170/171

**Review by ThomasWaldmann — CHANGES_REQUESTED** [review]
looks very good, 2 minor things i noticed.

**Review by ThomasWaldmann — APPROVED** [review]
LGTM!

**Comment by bigtedde** [comment]
> LGTM!

Squashed and ready for you to merge 👍

**Comment by ThomasWaldmann** [comment]
Thanks!

**Pull request merged**